services/src/main/java/io/scalecube/services/methods/ServiceMethodRegistryImpl.java

```diff
@@ -36,10 +42,14 @@ public void registerService(
                                   Reflect.requestType(method));
 
                           // register new service method invoker
-                          methodInvokers.put(
-                              methodInfo.qualifier(),
-                              new ServiceMethodInvoker(
-                                  method, serviceInstance, methodInfo, errorMapper, dataDecoder));
+                          String qualifier = methodInfo.qualifier();
+                          ServiceMethodInvoker invoker = new ServiceMethodInvoker(
+                                  method, serviceInstance, methodInfo, errorMapper, dataDecoder);
+
+                          if (methodInvokers.containsKey(qualifier)) {
+                            LOGGER.warn("<{}> for API <{}> will be replaced by the following methodInvoker: <{}>", methodInvokers.get(qualifier), qualifier, invoker);
+                          }
```

Comment:
Throw  exception (`IllegalStateException` should be fine) instead of allowing client to progress with wrong state.